src/integration-test/java/com/commercetools/sync/integration/externalsource/states/StateSyncIT.java

```diff
@@ -943,7 +744,10 @@ void sync_WithUpdatedTransitionAndClientThrowsError_ShouldAddErrorMessage() {
             })
             .warningCallback((exception, newResource, oldResource) ->
                 warningCallBackMessages.add(exception.getMessage()))
-            .build();
+            .build());
+
+        when(stateSyncOptions.getCtpClient()).thenReturn(spyClient);
```

Comment:
I assume some integration tests in the StateSyncIT could be moved to unit tests, any reason to no move? 

`sync_withCreateStateException_shouldPrintMessage`
`sync_withNewStateWihCreationException_shouldPrintErrorMessage`
`sync_WithExceptionWhenFetchingUnresolvedTransition_ShouldPrintErrorMessage`
`sync_WithUpdatedTransitionAndClientThrowsError_ShouldAddErrorMessage`

Actually those looks like a unit test becasue of mocking execute of client .

Reply:
`sync_withCreateStateException_shouldPrintMessage`
`sync_withNewStateWihCreationException_shouldPrintErrorMessage`
- For creation resources, the exception has been handled in `BaseService` class.  
https://github.com/commercetools/commercetools-sync-java/blob/471f863328aafbcb9326be171ae97ae64d60e30b/src/main/java/com/commercetools/sync/services/impl/BaseService.java#L306
If we move them to unit test class, empty object is mocked instead of a throwable object when calling method of `StateServiceImpl`. My concern is that it is a bit confusing to developer, because no exception class is needed in test case, so i keep it in integration test.

`sync_WithExceptionWhenFetchingUnresolvedTransition_ShouldPrintErrorMessage`
- No way to inject `UnresolvedTransitionsServiceImpl` into `StateSync`, which is required to do mocking in unit test.

`sync_WithUpdatedTransitionAndClientThrowsError_ShouldAddErrorMessage`
- I tried to move it but failed. When inputting three drafts together into sync process, the behaviour between IT and UT is different. In IT, the mocked exception can interrupt the entire sync process and quit. But In UT, the mocked exception does not affect sync process of other two drafts, and they continue to sync. Probably we can create follow-up ticket in future to investigate this question.

Reply:
Good to know the challenges, it looks like the effort needed for these changes does not needed at the moment, maybe create an issue for the descriptions here and put our issue list, what do you think?

Reply:
Agree. I would create issue later on .